packages/react-animation/src/carousel/util/getCarouselSlideProgress.ts

```diff
@@ -0,0 +1,34 @@
+import { type CarouselContext, CarouselType } from '../hooks/useCarousel.js';
+
+/**
+ * Returns a number between 0 and 1 representing the progress of the target slide.
+ *
+ * -1 represents the slide being completely off the left side of the carousel.
+ * 1 represents the slide being completely off the right side of the carousel.
+ * 0 represents the slide being at the left side of the carousel.
+ */
```

Comment:
We probably should have a few definitions and examples to clarify this better.

For example:
- what does "the carousel" mean in this context? Is it the visible bounding box where the carousel sits on the screen (e.g. in a container, that is often full-bleed, or could have paddings)
- or is it the focussed around a single "focus" slot that would be marked as a "center" one
- or something else

And then how that would impact the progress, I presume the following, in a carousel where 3 items would fit on screen:
- if a slide moves from the left side off the screen to it's left position, that would move for example 300px from -1 to 0.
- then if the slide would continue moving to the right, it would travel the same distance each time, but instead go from 0 to 0.33 (position 2), then 0.33 to 0.66 (position 3), and finally to progress 1 (off screen).

If the above is true, this would also mean that if you resize your screen, and suddenly 4 items would fit in the carousel, the progress of a slide at position one would be updated from 0.33 to 0.25, even though it wouldn't move.

I'm not saying this is bad, it's just that this isn't clearly documented yet, and might be important to understand to make use of it.